File: lc/1.two-sum.cpp
#include <bits/stdc++.h>
using namespace std;
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        vector<int> ans;
        unordered_map<int, int> m;
        for (size_t i = 0; i < nums.size(); i++){
            int t = target - nums[i];
            if (m.contains(t)){
                return {(int) i , m[t]};
            }
            else {
                m[nums[i]] = i;
            }
        }
        return {};
    }
};
// @lc code=end

File: lc/1.two-sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_to_index = {}
        for i in range(len(nums)):
            num_to_index[nums[i]] = i
        for i in range(len(nums)):
            target_num = target - nums[i]
            if target_num == nums[i]:
                if num_to_index[target_num] != i:
                    return [i, num_to_index[target_num]]
            elif target_num in num_to_index:
                return [i, num_to_index[target_num]]
        return []
